tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -621,8 +621,9 @@ public synchronized void serviceInit(final Configuration conf) throws Exception
       }
     }
 
-    rawExecutor = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setDaemon(true)
-        .setNameFormat("App Shared Pool - " + "#%d").build());
+    Executors.newFixedThreadPool(conf.getInt(TezConfiguration.TEZ_AM_DAG_APPCONTEXT_THREAD_COUNT_LIMIT,
```

Comment:
This thread pool is dangling. I think this was supposed to be assigned to `rawExecutor`?

Reply:
good catch, fixed